lib/ansible/modules/system/parted.py

```diff
@@ -0,0 +1,630 @@
+short_description: Configure block device partitions
+version_added: "2.3"
+description:
+  - This module allows configuring block device partition using the C(parted)
+    command line tool. For a full description of the fields and the options
+    check the GNU parted manual.
+notes:
+  - When fetching information about a new disk and when the version of parted
+    installed on the system is before version 3.1, the module queries the kernel
+    through C(/sys/) to obtain disk information. In this case the units CHS and
+    CYL are not supported.
+requirements:
+  - This module requires either parted version >= 3.1. If parted version < 3.1
+    the it is required a Linux version running the sysfs file system C(/sys/).
+options:
+  device:
+    description: The block device (disk) where to operate.
+    required: True
+  align:
+    description: Set alignment for newly created partitions.
+    choices: ['none', 'cylinder', 'minimal', 'optimal']
+    required: False
+    default: optimal
+  number:
+    description:
+     - The number of the partition to work with or the number of the partition
+       that will be created. Required when performing any action on the disk,
+       except fetching information.
+    required: False
+  unit:
+    description:
+     - Selects the current default unit that Parted will use to display
+       locations and capacities on the disk and to interpret those given by the
+       user if they are not suffixed by an unit. When fetching information about
+       a disk, it is always recommended to specify a unit.
+    choices: [
+       's', 'B', 'KB', 'KiB', 'MB', 'MiB', 'GB', 'GiB', 'TB', 'TiB', '%', 'cyl',
+       'chs', 'compact'
+    ]
+    required: False
+    default: KiB
+  label:
+    description: Creates a new disk label.
+    choices: [
+       'aix', 'amiga', 'bsd', 'dvh', 'gpt', 'loop', 'mac', 'msdos', 'pc98',
+       'sun', ''
+    ]
+    required: False
+    default: msdos
+  part_type:
+    description:
+     - Is one of 'primary', 'extended' or 'logical' and may be specified only
+       with 'msdos' or 'dvh' partition tables. A name must be specified for a
+       'gpt' partition table. Neither part-type nor name may be used with a
+       'sun' partition table.
+    choices: ['primary', 'extended', 'logical']
+    required: False
+  part_start:
+    description:
+     - Where the partition will start as offset from the beginning of the disk,
+       that is, the "distance" from the start of the disk.
+    required: False
+    default: 0%
+  part_end :
+    description:
+     - Where the partition will end as offset from the beginning of the disk,
+       that is, the "distance" from the start of the disk.
+    required: False
+    default: 100%
+  name:
+    description:
+     - Sets the name for the partition number (GPT, Mac, MIPS and PC98 only).
+    required: False
+  flags:
+    description: A list of the flags that has to be set on the partition.
+    required: False
+    default: null
```

Comment:
Parameters that are not required, don't need a `required: False`, also `default: null` is not needed here.
This is implied.